app/controllers/DataSetController.scala

```diff
@@ -56,7 +56,8 @@ class DataSetController @Inject()(userService: UserService,
       (__ \ 'displayName).readNullable[String] and
       (__ \ 'sortingKey).readNullable[Long] and
       (__ \ 'isPublic).read[Boolean] and
-      (__ \ 'tags).read[List[String]]).tupled
+      (__ \ 'tags).read[List[String]] and
+      (__ \ 'folderId).read[ObjectId]).tupled
```

Comment:
make nullable

Reply:
double check that the frontend does not send very old values, loaded long before the move

Reply:
> make nullable

@fm3 I did not find a change for that, but I might have missed it.

> double check that the frontend does not send very old values, loaded long before the move

Was this checked? @fm3 @philippotto

Reply:
Good catch, I missed this. I now pushed a change where the backend dataset update route expects the folderId as nullable, and does not change the folderId if no new one is passed.

@philippotto Jonathan pointed out that when multiple users edit a dataset (or the same user in multiple tabs), this could lead to old values being re-used. Assume in tab A the dashboard is open, in tab B the dataset settings. In tab B the dataset settings are changed (say, isPublic), then, hours later, in tab A, the dataset is dragged to a new folder, triggering a new update request. This would set isPublic to false again, right? We thinkt he frontend should probably guard against this by fetching the dataset info again before updating it. Or alternatively, the fields should all be nullable and changed only if sent. For that we would of course need a new way of encoding setting stuff no None, e.g. the description.

Reply:
Good point! I think, we have this problem potentially at other places, too, and it existed before for datasets in the dashboard as well (e.g., when editing tags). However, the issue is likely exacerbated now due to the folder property of datasets.

> Or alternatively, the fields should all be nullable and changed only if sent. For that we would of course need a new way of encoding setting stuff no None, e.g. the description. What do you think?

I think, this would be the best way. It doesn't solve all problems automatically (e.g., the tags are a single property of datasets where conflicting writes could also lose earlier updates), but it's a very good measure to avoid that unrelated change sets disturb each other. This is also how idiomatic, immutable updates in javascript are often done.

Differentiating between "should not be updated" and "should be set to null" is usually done by checking whether the key exists at all. So, this update payload `{description: null}` would remove the description, while this update payload `{tags: ["a tag"]}` wouldn't touch the description.

Is this possible in scala? If so, this would be great. Then, I can update the update-code in the front-end.

Additionally, regular polling would probably be a good idea, too, but I'd need to take care to not introduce janky updates to the lists.

> We thinkt he frontend should probably guard against this by fetching the dataset info again before updating it.

I'm not really fond of the added round trip and would consider this rather as a last resort.

Reply:
> Differentiating between "should not be updated" and "should be set to null" is usually done by checking whether the key exists at all.

Hmm, I’m afraid there is no direct way to represent this in play/scala  – compare https://discuss.lightbend.com/t/distinguishing-between-omitted-and-null-values/7105

I think there are a few options
 - add something like `updateDescription: bool` to the parameters for all nullable fields. If this is true, then the value in the db should be set to the value passed in `description`, be it null or something else
 - create different routes for each field, call only those that need changing
 - use fully custom deserialization in the backend, and use double wrapped `Option[Option[String]]` to represent the different states, compare https://stackoverflow.com/questions/48510822/deserialize-json-distinguising-missing-and-null-values
 - do the roundtrip as indicated before

My personal preference at this time would be the first variant using booleans. What do you think? cc @jstriebel

Reply:
(1) and (2) sound a bit tedious as an interface to be honest. (3) would be fine with me, but it's probably tedious to implement, too.

Could the backend list the available keys of the passed JsObject to know whether a `None` value was actually listed or not? That would be akin to passing booleans (because the is-key-included-in-key-list check would return a boolean, too), but without explicitly passing these booleans?

Reply:
I agree that all of the options are not super convincing. Would it be possible to add a custom type, e.g. Unset vs Set, which would be similar to an Option, and Unset is the default? We use something similar in the python libs:
https://github.com/scalableminds/webknossos-libs/blob/master/webknossos/webknossos/client/_generated/models/current_user_info_response_200.py#L36
Or would that also involve custom deserialization? I'd hope that there's a way to register custom deserialization hooks?

Reply:
A custom wrapper type is an interesting Idea, could be used just for such update requests. I’ll have a look.
The other question is whether this should block this PR. The problem certainly existed before folders, as philipp has already stated. I’d vote to make this a follow-up issue, what do you think?

Reply:
> I’d vote to make this a follow-up issue, what do you think?

Sound's fair for me. I'd say this issue is more pressing now, since different views (that easily are opened in separate tabs and for different users) now provoke this problem, and not just editing a single UI view in parallel. But it does not need to block this PR :+1:

Reply:
I’m afraid even this was already the case before (I believe tags can be added in the dashboard) 😅 

In either case, I created the issue https://github.com/scalableminds/webknossos/issues/6657

Reply:
Just for the record: I implemented a polling mechanism so that the datasets are updated regularly and automatically. Therefore, the chance of conflicting updates is greatly reduced, since the updates would need to be in the same time window (I think it's 60 seconds).